Classes/MapConverters/StepManiaMapConverter.cs

```diff
@@ -0,0 +1,340 @@
+    }
+
+    /// <summary>
+    /// MUSIC - copy as song.ogg into the beatmap folder.
+    /// .sm and .ssc also support .mp3 - in this case we skip copying it over
+    /// </summary>
+    private void ParseSong(RagnarockMap beatmap, SmFileExtra smFile)
+    {
+        String songFilename = smFile[SmFileAttribute.MUSIC];
+        string songURL = beatmap.PathOf(BeatmapDefaults.SongFilename);
+        if (songFilename != string.Empty && Path.GetExtension(songFilename) == ".ogg")
+        {
+            // can't copy over an existing file
+            if (File.Exists(songURL))
+            {
+                File.Delete(songURL);
+            }
+            File.Copy(Path.Combine(smFile.Directory, songFilename), songURL);
+        }
+    }
+
+    /// <summary>
+    /// BANNER - this one looks most likely to be supported as cover, but the aspect ratio might be wrong - 256x80 is the recommended size in ArrowVortex, while Ragnarock needs a square cover image.
+    /// In this case, I hope it will be recognisable from the preview after loading the map, otherwise it might be good to show a warning somewhere.
+    /// </summary>
+    private void ParseCover(RagnarockMap beatmap, SmFileExtra smFile)
+    {
+        string coverImageFilename = smFile[SmFileAttribute.BANNER];
+        if (coverImageFilename != string.Empty)
+        {
+            string coverUrl = beatmap.PathOf(coverImageFilename);
+            // can't copy over an existing file
+            if (File.Exists(coverUrl))
+            {
+                File.Delete(coverUrl);
+            }
+            File.Copy(Path.Combine(smFile.Directory, coverImageFilename), coverUrl);
+            beatmap.SetValue("_coverImageFilename", coverImageFilename);
+        }
+    }
+
+    /// <summary>
+    /// PREVIEW should take precedence over SAMPLESTART, SAMPLELENGTH according to .ssc documentation, but it isn't supported by ArrowVortex and StepmaniaUtils.<br/>
+    /// SAMPLESTART, SAMPLELENGTH - if both are provided, generate preview file based on those settings. We also need to have the music file already.
+    /// </summary>
+    private void ParsePreview(RagnarockMap beatmap, SmFileExtra smFile)
+    {
+        string songURL = beatmap.PathOf((string)beatmap.GetValue("_songFilename"));
+        if (double.TryParse(smFile[SmFileAttribute.SAMPLESTART], CultureInfo.InvariantCulture, out double sampleStart) &&
+            double.TryParse(smFile[SmFileAttribute.SAMPLELENGTH], CultureInfo.InvariantCulture, out double sampleLength) &&
+            File.Exists(songURL))
+        {
+            string saveURL = beatmap.PathOf("preview.ogg");
+            int startMin = (int)(sampleStart / 60);
+            int startSec = (int)(sampleStart - 60 * startMin);
+            int endMin = (int)((sampleStart + sampleLength) / 60);
+            int endSec = (int)(sampleStart + sampleLength - 60 * endMin);
+            // TODO: for some reason, two windows open for ffmpeg and they clash with each other.
```

Comment:
I'm not sure why, but it seems like the process starts twice here. Preview is still generated correctly, but there's a nasty error that the ffmpeg_temp.exe couldn't be deleted, because the second process was still using it.

Reply:
that's my bad - Helper.cs line 145 (`p.Start();`) should be commented out.

Reply:
Downmerged the change and retested. It's all good now.